tecken/dockerflow_extra.py

```diff
@@ -57,44 +57,34 @@ def check_url(url, setting_key):
         bucket = StorageBucket(url)
         if not bucket.private:
             return
-        if bucket.is_google_cloud_storage:
-            try:
-                bucket.client.get_bucket(bucket.name)
-                checked.append(url)
-            except google_BadRequest as exception:
+        try:
+            if not bucket.exists():
                 errors.append(
                     checks.Error(
-                        f"Unable to connect to {url} (bucket={bucket.name!r}, "
-                        f"because bucket not found due to {exception}",
+                        f"Unable to connect to {url} (bucket={bucket.name!r}), "
+                        f"because bucket not found",
                         id="tecken.health.E003",
                     )
                 )
-        else:
-            try:
-                bucket.client.head_bucket(Bucket=bucket.name)
-            except ClientError as exception:
-                if exception.response["Error"]["Code"] in ("403", "404"):
-                    errors.append(
-                        checks.Error(
-                            f"Unable to connect to {url} (bucket={bucket.name!r}, "
-                            f"found in settings.{setting_key}) due to "
-                            f"ClientError ({exception.response!r})",
-                            id="tecken.health.E002",
-                        )
-                    )
-                else:
-                    raise
-            except EndpointConnectionError:
-                errors.append(
-                    checks.Error(
-                        f"Unable to connect to {url} (bucket={bucket.name!r}, "
-                        f"found in settings.{setting_key}) due to "
-                        f"EndpointConnectionError",
-                        id="tecken.health.E001",
-                    )
+        except (GoogleAPIError, ClientError) as exception:
+            errors.append(
+                checks.Error(
+                    f"Unable to connect to {url} (bucket={bucket.name!r}), "
+                    f"due to {type(exception).__name__}: {exception}",
+                    id="tecken.health.E002",
+                )
+            )
+        except EndpointConnectionError:
+            errors.append(
+                checks.Error(
+                    f"Unable to connect to {url} (bucket={bucket.name!r}, "
+                    f"found in settings.{setting_key}) due to "
+                    f"EndpointConnectionError",
+                    id="tecken.health.E001",
```

Comment:
Can't `.exists()` throw an exception that wraps the actual exception? Then we don't need the S3 and GCP specific handling here.

Reply:
Yes! Python 3's exception chaining makes this nice for uncaught exceptions. For the dockerflow_extra checks, I'll need to get some details into the error message, but I think I can copy the client's exception string, rather than wrap the exception.